app/src/main/java/net/gsantner/opoc/util/StringUtils.java

```diff
@@ -623,75 +625,106 @@ public void clear() {
+        @Override
+        public void clearSpans() {
+            // Do nothing
+        }
+
+        @Override
+        public void setFilters(InputFilter[] filters) {
+            // Do nothing
+        }
+
+        @Override
+        public InputFilter[] getFilters() {
+            return null;
         }
 
         @Override
         public void setSpan(Object what, int start, int end, int flags) {
-            select().setSpan(what, start, end, flags);
+            // Do nothing
         }
 
         @Override
         public void removeSpan(Object what) {
-            select().removeSpan(what);
+            // Do nothing
         }
 
         @Override
         public <T> T[] getSpans(int start, int end, Class<T> type) {
-            return select().getSpans(start, end, type);
+            return null;
         }
 
         @Override
         public int getSpanStart(Object tag) {
-            return select().getSpanStart(tag);
+            return -1;
         }
 
         @Override
         public int getSpanEnd(Object tag) {
-            return select().getSpanEnd(tag);
+            return -1;
         }
 
         @Override
         public int getSpanFlags(Object tag) {
-            return select().getSpanFlags(tag);
+            return 0;
         }
 
         @Override
         public int nextSpanTransition(int start, int limit, Class type) {
-            return select().nextSpanTransition(start, limit, type);
+            return -1;
         }
+    }
 
-        @Override
-        public int length() {
-            return select().length();
-        }
+    public static Runnable makeDebounced(final long delayMs, final Runnable callback) {
+        return makeDebounced(null, delayMs, callback);
+    }
 
-        @Override
-        public char charAt(int index) {
-            return select().charAt(index);
-        }
+    // Debounce any callback
+    public static Runnable makeDebounced(final Handler handler, final long delayMs, final Runnable callback) {
+        final Handler _handler = handler == null ? new Handler(Looper.getMainLooper()) : handler;
+        final Object sync = new Object();
+        return () -> {
+            synchronized (sync) {
+                _handler.removeCallbacks(callback);
+                _handler.postDelayed(callback, delayMs);
+            }
+        };
+    }
 
-        @NonNull
-        @Override
-        public CharSequence subSequence(int start, int end) {
-            return select().subSequence(start, end);
+    // Converts region to string with a minimum of work
+    public static String toString(final CharSequence source, int start, int end) {
+        if (source instanceof String) {
+            // Already very fast
+            return ((String) source).substring(start, end);
         }
+
+        final char[] buf = new char[end - start];
+        TextUtils.getChars(source, start, end, buf, 0);
+        return new String(buf);
```

Comment:
`editable.subSequence(start, end).toString()` copies spans etc. This avoids that extra work